### PR TITLE
[REVIEW] Restructure Movielens  inference nbs

### PR DESCRIPTION
**Pull request by rnyak**

### CODE REVIEW

**Comment by rnyak** [comment]
@benfred I am getting error from HugeCTR inference stage, if I use the same workflow from multi-hot ETL.

```
export_hugectr_ensemble(workflow=workflow, 
                        hugectr_model_path="/model/movielens_hugectr/1/",
                        hugectr_params=hugectr_params,
                        name="movielens", 
                        output_path="/model/models/", 
                        label_columns=LABEL_COLUMNS, 
                        cats=CATEGORICAL_COLUMNS,
                        max_batch_size=64)
````

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-10-9864399f9e53> in <module>
      6 hugectr_params["embedding_vector_size"] = 16
      7 hugectr_params["n_outputs"] = 1
----> 8 export_hugectr_ensemble(workflow=workflow, 
      9                         hugectr_model_path="/model/movielens_hugectr/1/",
     10                         hugectr_params=hugectr_params,
/nvtabular/nvtabular/inference/triton/__init__.py in export_hugectr_ensemble(workflow, hugectr_model_path, hugectr_params, name, output_path, label_columns, version, cats, conts, max_batch_size)
    121     # generate the nvtabular triton model
    122     preprocessing_path = os.path.join(output_path, name + "_nvt")
--> 123     nvt_config = generate_nvtabular_model(
    124         workflow=workflow,
    125         name=name + "_nvt",
/nvtabular/nvtabular/inference/triton/__init__.py in generate_nvtabular_model(workflow, name, output_path, version, output_model, cats, conts, max_batch_size)
    173 
    174     workflow.save(os.path.join(output_path, str(version), "workflow"))
--> 175     config = _generate_nvtabular_config(
    176         workflow, name, output_path, output_model, max_batch_size, cats, conts
    177     )
/nvtabular/nvtabular/inference/triton/__init__.py in _generate_nvtabular_config(workflow, name, output_path, output_model, max_batch_size, cats, conts)
    258     if output_model == "hugectr":
    259         for column in workflow.column_group.input_column_names:
--> 260             dtype = workflow.input_dtypes[column]
    261             config.input.append(
    262                 model_config.ModelInput(name=column, data_type=_convert_dtype(dtype), dims=[-1])
KeyError: 'genres'
```

**Comment by rnyak** [comment]
@benfred can we merge these notebooks? HugeCTR inference is giving an error, I am gonna create a bug issue.

File: examples/getting-started-movielens/03a-Training-with-TF.ipynb [hunk 1]
```diff
@@ -635,6 +633,26 @@
...
+     "text": [
+      "INFO:tensorflow:Assets written to: /model/movielens_tf/1/model.savedmodel/assets\n"
+     ]
+    }
+   ],
+   "source": [
+    "MODEL_NAME_TF = os.environ.get(\"MODEL_NAME_TF\", \"movielens_tf\")\n",
+    "MODEL_PATH_TEMP_TF = os.path.join(MODEL_BASE_DIR, MODEL_NAME_TF, \"1/model.savedmodel\")\n",
+    "\n",
+    "model.save(MODEL_PATH_TEMP_TF)"
```
Comment (benfred) [inline examples/getting-started-movielens/03a-Training-with-TF.ipynb:654]: We should base this off of INPUT_DATA_DIR so that the unittests can pass

**Review by benfred — COMMENTED** [review]
Lets get this test passing: ```pytest tests/unit/test_notebooks.py::test_movielens_example```

File: examples/getting-started-movielens/01-Download-Convert.ipynb [hunk 2]
```diff
@@ -89,6 +89,26 @@
...
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "downloading ml-25m.zip: 262MB [00:07, 35.2MB/s]                            \n",
+      "unzipping files: 100%|██████████| 8/8 [00:09<00:00,  1.20s/files]\n"
+     ]
+    }
+   ],
+   "source": [
+    "from nvtabular.utils import download_file\n",
```
[the remark this answers is not in the document]
Comment (rnyak) [reply examples/getting-started-movielens/01-Download-Convert.ipynb:101]: I don't see a repeated line at my end.. strange.

[the remark this answers is not in the document]
Comment (benfred) [reply examples/getting-started-movielens/01-Download-Convert.ipynb:101]: It's at the end of the first block of imports at the top of the file

File: examples/getting-started-movielens/03a-Training-with-TF.ipynb (hunk 1 above)
Comment (rnyak) [reply examples/getting-started-movielens/03a-Training-with-TF.ipynb:654]: done

**Comment by rnyak** [comment]
@benfred I fixed model paths, removed `triton-inference` folder, moved `inference-HugeCtR` under the `getting started with movielens` folder. Not sure why Jenkins is failing for README.

File: examples/getting-started-movielens/03a-Training-with-TF.ipynb [hunk 3]
```diff
@@ -566,15 +567,11 @@
...
-      ]
-     },
-     "execution_count": 18,
-     "metadata": {},
-     "output_type": "execute_result"
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "('Failed to import pydot. You must `pip install pydot` and install graphviz (https://graphviz.gitlab.io/download/), ', 'for `pydotprint` to work.')\n"
+     ]
```
Comment (benfred) [inline examples/getting-started-movielens/03a-Training-with-TF.ipynb:574]: Should we re-render with pydot installed?

**Review by benfred — APPROVED** [review]

File: examples/getting-started-movielens/03a-Training-with-TF.ipynb (hunk 3 above)
Comment (rnyak) [reply examples/getting-started-movielens/03a-Training-with-TF.ipynb:574]: @benfred thanks will do asap.

**Pull request merged**